src/runQueryCommand.ts

```diff
@@ -24,7 +24,21 @@ export default (
       'X-Fauna-Source': 'VSCode'
     }
   });
-  const code = activeTextEditor.document.getText();
+
+  const selection = activeTextEditor.selection;
+  const selectedText = activeTextEditor.document.getText(selection);
+  let code = '';
+  if (selectedText.length > 0) {
+    code = selectedText;
+  } else {
+    code = activeTextEditor.document.getText();
+  }
```

Comment:
```suggestion
  const fql = selectedText.length > 0 ? selectedText : activeTextEditor.document.getText();
```

Reply:
We can use a ternary here to handle both the variable assignment and the conditional logic. Let's also update the variable name to be more clear.